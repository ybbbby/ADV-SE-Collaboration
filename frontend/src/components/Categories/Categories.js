import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import {
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Divider,
} from '@material-ui/core'
import FavoriteIcon from '@material-ui/icons/Favorite'
import HistoryIcon from '@material-ui/icons/History'
import FormatListNumberedIcon from '@material-ui/icons/FormatListNumbered'
import { Link } from 'react-router-dom'

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    padding: 0,
    marginTop: theme.spacing(3),
    boxShadow:
      '0px 0px 0px -1px rgba(0,0,0,0.2), -2px -1px 1px -1px rgba(0,0,0,0.14), 0px 1px 4px 1px rgba(0,0,0,0.12)',
  },
  link: {
    textDecoration: 'none',
    color: 'black',
  },
}))

export default function Categories() {
  const classes = useStyles()

  return (
    <List component="nav" className={classes.root}>
      <Link to="/" className={classes.link}>
        <ListItem button>
          <ListItemIcon>
            <FavoriteIcon />
          </ListItemIcon>
          <ListItemText primary="Favourites" />
        </ListItem>
      </Link>
      <Divider />
      <Link to="/eventsnearby/togo" className={classes.link}>
        <ListItem button>
          <ListItemIcon>
            <FormatListNumberedIcon />
          </ListItemIcon>
          <ListItemText primary="To go" />
        </ListItem>
      </Link>
      <Divider />
      <Link to="/eventsnearby/history" className={classes.link}>
        <ListItem button>
          <ListItemIcon>
            <HistoryIcon />
          </ListItemIcon>
          <ListItemText primary="History" />
        </ListItem>
      </Link>
    </List>
  )
}
