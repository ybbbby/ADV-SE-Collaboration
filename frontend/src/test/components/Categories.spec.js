/* eslint-env jest */
import { shallow } from 'enzyme'
import React from 'react'
import { HashRouter as Router } from 'react-router-dom'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Categories from '../../components/Categories/Categories'

describe('ShareModal', () => {
  test('render logged in', () => {
    render(
      <Router>
        <Categories isLogin={true} />
      </Router>
    )
    expect(screen.queryByText('logout')).toBeDefined()
  })

  test('render not logged in', () => {
    render(
      <Router>
        <Categories isLogin={false} />
      </Router>
    )
    expect(screen.queryByText('logout')).toBeNull()
  })

  test('click logout button (success)', async () => {
    fetch.mockResponseOnce('')
    render(
      <Router>
        <Categories isLogin={true} />
      </Router>
    )
    await waitFor(() => userEvent.click(screen.getByText('Logout')))
    // logout logic tested in api tests
  })

  test('click logout button (fail)', async () => {
    fetch.mockReject(() => Promise.reject('API is down'))
    render(
      <Router>
        <Categories isLogin={true} />
      </Router>
    )
    await waitFor(() => userEvent.click(screen.getByText('Logout')))
    // logout logic tested in api tests
  })

  test('close login modal when clicks the close button', async () => {
    render(
      <Router>
        <Categories isLogin={false} />
      </Router>
    )
    userEvent.click(screen.getByText('History'))
    userEvent.click(screen.getByRole('button', { name: 'close' }))
    // the handleClose function is tested in the LoginModal tests
  })

  test('renders login modal when clicks the list item', () => {
    render(
      <Router>
        <Categories isLogin={false} />
      </Router>
    )
    userEvent.click(screen.getByText('To go'))
    expect(screen.findAllByRole('presentation')).toBeDefined()
  })

  test('clicks the list item when logged in', () => {
    render(
      <Router>
        <Categories isLogin={true} />
      </Router>
    )
    userEvent.click(screen.getByText('To go'))
    expect(screen.queryByText('Welcome Back!')).toBeNull()
  })

  it('snapshot when logged in', () => {
    const wrapper = shallow(<Categories isLogin={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot when not logged in', () => {
    const wrapper = shallow(<Categories isLogin={false} />)
    expect(wrapper).toMatchSnapshot()
  })
})
