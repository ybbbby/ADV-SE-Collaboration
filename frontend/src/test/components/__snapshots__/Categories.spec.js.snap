// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareModal snapshot when logged in 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-27"
  component="nav"
>
  <Link
    className="makeStyles-link-28"
    to="/"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Home"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="/events/liked"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Favourites"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="/events/ongoing"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="To go"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="/events/history"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="History"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(ListItem))
    button={true}
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemIcon))>
      <Memo([object Object]) />
    </WithStyles(ForwardRef(ListItemIcon))>
    <WithStyles(ForwardRef(ListItemText))
      primary="Logout"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;

exports[`ShareModal snapshot when not logged in 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-27"
  component="nav"
>
  <Link
    className="makeStyles-link-28"
    to="#"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Home"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="#"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Favourites"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="#"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="To go"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <WithStyles(ForwardRef(Divider)) />
  <Link
    className="makeStyles-link-28"
    to="#"
  >
    <WithStyles(ForwardRef(ListItem))
      button={true}
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <Memo([object Object]) />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="History"
      />
    </WithStyles(ForwardRef(ListItem))>
  </Link>
  <LoginModal
    handleClose={[Function]}
    open={false}
  />
</WithStyles(ForwardRef(List))>
`;
